src/twisted/internet/test/test_base.py

```diff
@@ -155,6 +158,32 @@ def fakeGetHostByName(name):
         result.put(IOError("The I/O was errorful"))
 
 
+    def test_resolverGivenBytes(self):
+        """
+        L{ThreadedResolver.getHostByName} is passed L{bytes}.
+        """
+        calls = []
+
+        @implementer(IResolverSimple)
+        class FakeResolver(object):
+            def getHostByName(self, name, timeouts=()):
+                calls.append(name)
+                return Deferred()
+
+        class JustEnoughReactor(ReactorBase):
+            def installWaker(self):
+                pass
+
+        fake = FakeResolver()
+        reactor = JustEnoughReactor()
+        reactor.installResolver(fake)
+        rec = FirstOneWins(Deferred())
+        reactor.nameResolver.resolveHostName(rec, u"xn--pxaix.localhost")
+
+        self.assertEqual(len(calls), 1)
+        self.assertIs(type(calls[0]), bytes)
```

Comment:
Given that the interface of `IResolverSimple` (a) is formally documented as taking `str` and (b) is _practically_ "whatever `gethostbyname` in the stdlib takes" (which is to say: native `str`), this should _really_ be asserting that it is native `str` both on Python 2 and 3.